packages/core/lib/commands/init/initSource/truffle-config.js

```diff
@@ -48,28 +48,37 @@ module.exports = {
     // },
     // Another network with more advanced options...
     // advanced: {
-    // port: 8777,             // Custom port
-    // network_id: 1342,       // Custom network
-    // gas: 8500000,           // Gas sent with each transaction (default: ~6700000)
-    // gasPrice: 20000000000,  // 20 gwei (in wei) (default: 100 gwei)
-    // from: <address>,        // Account to send txs from (default: accounts[0])
-    // websocket: true        // Enable EventEmitter interface for web3 (default: false)
+    //  port: 8777,            // Custom port
+    //  network_id: 1342,      // Custom network
+    //  gas: 8500000,          // Gas sent with each transaction (default: ~6700000)
+    //  gasPrice: 20000000000, // 20 gwei (in wei) (default: 100 gwei)
+    //  from: <address>,       // Account to send txs from (default: accounts[0])
+    //  websocket: true        // Enable EventEmitter interface for web3 (default: false)
     // },
     // Useful for deploying to a public network.
+    // You can create separate configurations for multiple public networks using different names, e.g. mainnet, ropsten, rinkeby.
+    // public: {
     // NB: It's important to wrap the provider as a function.
-    // ropsten: {
-    // provider: () => new HDWalletProvider(mnemonic, `https://ropsten.infura.io/v3/YOUR-PROJECT-ID`),
-    // network_id: 3,       // Ropsten's id
-    // gas: 5500000,        // Ropsten has a lower block limit than mainnet
-    // confirmations: 2,    // # of confs to wait between deployments. (default: 0)
-    // timeoutBlocks: 200,  // # of blocks before a deployment times out  (minimum/default: 50)
-    // skipDryRun: true     // Skip dry run before migrations? (default: false for public nets )
+    // NB: Please use a different provider URL for different public networks.For examples, 
+    // NB:   mainnet : `https://mainnet.infura.io/v3/YOUR-PROJECT-ID`
+    // NB:   rinkeby : `https://rinkeby.infura.io/v3/YOUR-PROJECT-ID`
+    // NB: Also, it is recommended to use the same mnemonic and use different indexes (0, 1, 2 ...) for different public networks.
+    //  provider: () => new HDWalletProvider(mnemonic, `https://network.io`, 0),
+    //  network_id: 1,                      // Network ID: mainnet => 1; ropsteb => 3; rinkeby => 4
+    // NB: You need to use type 2 transactions on the public network, otherwise you may get odd errors when interacting.
```

Comment:
I don't really think it is necessary yet to use type 2 transactions - I think you just won't benefit from these newer features, right?

Reply:
That's my understanding also...

Reply:
@wxy can you share an example of odd errors you encountered?

Reply:
@cds-amal @haltman-at @eggplantzzz 
I've mentioned it in the commit comments for this PR, but I don't have the original full log and error message anymore.

This is just a few suggestions and perhaps just my own personal experience of the problem. I thought it might be helpful to more people as well, hence submit this PR. Also, the trend is to use type 2 transactions anyway, so I would suggest changing to type 2. however, if this is officially deemed unnecessary, then I will close this PR.

Reply:
Also, for type 2 transactions, in addition to fixing the error I encountered, I think it has the extra benefit of using the gas/gasPrice parameter, where you may be paying more gas, and using maxFeePerGas/maxPriorityFeePerGas, which will use no more gas than the limit based on current network conditions.

Reply:
Thanks @wxy , I will discuss with the team. Preparing for the Type 2 transactions was a perspective I didn't consider. Thanks for raising it!